packages/amplify_core/lib/src/types/query/query_field_operators.dart

```diff
@@ -40,8 +40,18 @@ abstract class QueryFieldOperator<T> {
 
   const QueryFieldOperator(this.type);
 
+  /// Evaluates this QueryFieldOperator against [other],
+  /// where [other] is a field on the model.
   bool evaluate(T? other);
 
+  /// Similar to `evaluate`, except `other` should be the
+  /// *serialized* value of a field on the model.
+  ///
+  /// This should be used to support comparisons with models
+  /// that were generated prior to `toMap()` being added.
+  @Deprecated('To be removed in next major release')
```

Comment:
Should mention the action item inside the `@Deprecated` instead of a comment here so developer will know what to do seeing the deprecation message while building App?

Reply:
Updated.

I don't expect customers to see this since this is used internally, although it is technically part of the public API. I could make it internal, but then it makes it a pain to use outside of amplify_core. Updating the deprecation message in case customers stumble upon this API seems reasonable 👍